From: Jean Delvare <khali@linux-fr.org>
Date: Tue, 13 Feb 2007 21:08:59 +0000 (+0100)
Subject: i2c-piix4: Add support for the ATI SB600
X-Git-Tag: v2.6.21~1021^2~4^2~14
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=4e6697fcc194db8b45559a9863947c6cbfeea363

i2c-piix4: Add support for the ATI SB600

Add support for the ATI SB600 SMBus controller.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/Documentation/i2c/busses/i2c-piix4 linux-source-2.6.18/Documentation/i2c/busses/i2c-piix4
--- linux-source-2.6.18.orig/Documentation/i2c/busses/i2c-piix4	2006-09-19 21:42:06.000000000 -0600
+++ linux-source-2.6.18/Documentation/i2c/busses/i2c-piix4	2007-08-03 02:39:13.196906208 -0600
@@ -6,7 +6,7 @@ Supported adapters:
     Datasheet: Publicly available at the Intel website
   * ServerWorks OSB4, CSB5, CSB6 and HT-1000 southbridges
     Datasheet: Only available via NDA from ServerWorks
-  * ATI IXP southbridges IXP200, IXP300, IXP400
+  * ATI IXP200, IXP300, IXP400 and SB600 southbridges
     Datasheet: Not publicly available
   * Standard Microsystems (SMSC) SLC90E66 (Victory66) southbridge
     Datasheet: Publicly available at the SMSC website http://www.smsc.com
diff -urpN linux-source-2.6.18.orig/drivers/i2c/busses/Kconfig linux-source-2.6.18/drivers/i2c/busses/Kconfig
--- linux-source-2.6.18.orig/drivers/i2c/busses/Kconfig	2007-05-23 08:49:59.000000000 -0600
+++ linux-source-2.6.18/drivers/i2c/busses/Kconfig	2007-08-03 02:39:13.196906208 -0600
@@ -176,6 +176,7 @@ config I2C_PIIX4
 	    ATI IXP200
 	    ATI IXP300
 	    ATI IXP400
+	    ATI SB600
 	    Serverworks OSB4
 	    Serverworks CSB5
 	    Serverworks CSB6
diff -urpN linux-source-2.6.18.orig/drivers/i2c/busses/i2c-piix4.c linux-source-2.6.18/drivers/i2c/busses/i2c-piix4.c
--- linux-source-2.6.18.orig/drivers/i2c/busses/i2c-piix4.c	2006-09-19 21:42:06.000000000 -0600
+++ linux-source-2.6.18/drivers/i2c/busses/i2c-piix4.c	2007-08-03 02:39:13.196906208 -0600
@@ -23,6 +23,7 @@
    Supports:
 	Intel PIIX4, 440MX
 	Serverworks OSB4, CSB5, CSB6, HT-1000
+	ATI IXP200, IXP300, IXP400, SB600
 	SMSC Victory66
 
    Note: we assume there can only be one device, with one SMBus interface.
@@ -396,6 +397,8 @@ static struct pci_device_id piix4_ids[] 
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP400_SMBUS),
 	  .driver_data = 0 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_IXP600_SMBUS),
+	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_OSB4),
 	  .driver_data = 0 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_SERVERWORKS, PCI_DEVICE_ID_SERVERWORKS_CSB5),
diff -urpN linux-source-2.6.18.orig/include/linux/pci_ids.h linux-source-2.6.18/include/linux/pci_ids.h
--- linux-source-2.6.18.orig/include/linux/pci_ids.h	2007-08-03 01:48:49.152931000 -0600
+++ linux-source-2.6.18/include/linux/pci_ids.h	2007-08-03 02:39:13.196906208 -0600
@@ -362,6 +362,7 @@
 #define PCI_DEVICE_ID_ATI_IXP400_SATA2	0x437a
 #define PCI_DEVICE_ID_ATI_IXP600_SATA	0x4380
 #define PCI_DEVICE_ID_ATI_IXP600_SRAID	0x4381
+#define PCI_DEVICE_ID_ATI_IXP600_SMBUS	0x4385
 #define PCI_DEVICE_ID_ATI_IXP600_IDE	0x438c
 #define PCI_DEVICE_ID_ATI_IXP700_SATA	0x4390
 #define PCI_DEVICE_ID_ATI_IXP700_IDE	0x439c
